pkg/config/config_test.go

```diff
@@ -980,6 +980,33 @@ packages:
                     recursive: true
                     with-expecter: false
 with-expecter: false
+`,
+		},
+		{
+			name: "test with excluded subpackage",
+			cfgYaml: `
+with-expecter: False
+dir: foobar
+packages:
+  github.com/vektra/mockery/v2/pkg/fixtures/example_project/pkg_with_subpkgs/subpkg2:
+    config:
+      recursive: True
+      with-expecter: True
+      all: True
+      exclude:
+        - subpkg3
+`,
+			wantCfgMap: `dir: foobar
```

Comment:
I understand about the funkiness of the test. That's okay, people can come back to this PR if they're confused :)